picodom.d.ts

```diff
@@ -0,0 +1,26 @@
+export interface VNode<D> {
+  tag: string
+  data?: D
+  children: VElement<{}>[]
```

Comment:
@pspeter3 I think we might not need VElement, but please correct me if I am wrong. The signature of a Component is good, but a virtual node children property is always an array. We make sure of that inside `h`.